### PR TITLE
Pod Security Policies

### PR DESCRIPTION
**Pull request by gambol99**
The current implementation doesn't work with PodSecurityPolicies admission controller enabled due to no psp policies. This PR adds a default psp policy for the kubelet users and and the kube-system namespace.

- added a default pod security policy covering the kube-system namespace
- this also covers the kubelet user as without this static pod are not longer visible, which effects the ability of the dns-controller to add the internal api dns records and thus no nodes can come up.

I still need to test this with 1.8.0; thus far only 1.7.2 / 1.7.7 has been tested

### CODE REVIEW

**Comment by gambol99** [comment]
i'm gonna leave this one as WIP for now and perhaps offline chat on it. The experience of PSP hasn't been get and at present a number of outstanding PR's targeted for a 1.8.x release to make this admission controller for user friendly.

**Comment by chrislovecnm** [comment]
@gambol99 is this something that you are working on? Just grooming old PRs

**Comment by gambol99** [comment]
apologizes for the delay @chrislovecnm ... currently away for xmas in italy so trying to stay away from a laptop :-) .. The PR was functional, the issue was due to problem with the admission controller itself. Effectively if you had a user or service account which via rbac can see all policies (i.e. pretty much most things deployed in kube-system as it was deployed via an cluster wide admin) the controller will randomly pick a policy. So there's a percentage of pods restarted which will end up with a policy which might conflict. In our case a few restarts of kube-dns would fail due to the RunAsNonRoot requirement. Their we two PR's which fix this behavior, one which orders the policy via [name](https://github.com/kubernetes/kubernetes/pull/52849) and the other which add's a [priority](https://github.com/kubernetes/kubernetes/pull/45358) field to the policies. Both are scheduled for a 1.9 release ..

File: pkg/apis/kops/v1alpha1/componentconfig.go [hunk 1]
```diff
@@ -255,6 +255,7 @@ type KubeAPIServerConfig struct {
 	// of the authorities in the oidc-ca-file
 	OIDCCAFile *string `json:"oidcCAFile,omitempty" flag:"oidc-ca-file"`
 	// The apiserver's client certificate used for outbound requests.
+	// The apiserver's client certificate used for outbound requests.
```
Comment (justinsb) [inline pkg/apis/kops/v1alpha1/componentconfig.go]: Duplicated line?

File: pkg/apis/kops/v1alpha1/componentconfig.go [hunk 2]
```diff
@@ -417,3 +417,14 @@ type CloudConfiguration struct {
 	VSphereDatastore     *string `json:"vSphereDatastore,omitempty"`
 	VSphereCoreDNSServer *string `json:"vSphereCoreDNSServer,omitempty"`
 }
+
+// HasAdmissionController checks if a specific admission controller is enabled
+func (c *KubeAPIServerConfig) HasAdmissionController(name string) bool {
```
Comment (justinsb) [inline pkg/apis/kops/v1alpha1/componentconfig.go]: I think to date we've avoided methods on the API objects themselves, but this one is simple enough that we can just see what happens :-)  I'm surprised we want it on anything other than the internal version (i.e. I'm surprised we want it on v1alpha1 and v1alpha2)

**Comment by gambol99** [comment]
@justinsb @chrislovecnm  ... since 1.9 this one is good to go as the issue effecting pods who's rbac permits seeing multiple policies has been fixed

File: upup/pkg/fi/cloudup/bootstrapchannelbuilder.go [hunk 3]
```diff
@@ -110,6 +110,27 @@ func (b *BootstrapChannelBuilder) buildManifest() (*channelsapi.Addons, map[stri
...
+		{
+			location := key + "/k8s-1.6.yaml"
+			id := "k8s-1.6"
+
+			addons.Spec.Addons = append(addons.Spec.Addons, &channelsapi.AddonSpec{
+				Name:              fi.String(key),
+				Version:           fi.String(version),
+				Selector:          map[string]string{"k8s-addon": key},
+				Manifest:          fi.String(location),
+				KubernetesVersion: ">=1.6.0",
```
Comment (gambol99) [inline upup/pkg/fi/cloudup/bootstrapchannelbuilder.go]: can i just change this to >=1.9.0 @justinsb

**Pull request merged**